### ISSUE TITLE
serverless: Try to bundle all dependencies for custom server

### ISSUE THREAD

**Issue by romainquellec** [zeit/next.js#6391]
# Feature request
At some point, we need to server next with a custom server and/or lambda.
When we build with target:serverless, lambda is not directly compatible with Serverless.

`const awsServerlessExpress = require("aws-serverless-express");` will crash the lambda (except special treatment)
(Maybe its already possible without making another webpack config)

## Describe the solution you'd like
Compile these files with next build. (next.config.js?)

## Additional context
https://github.com/romainquellec/boilerplate for example.

**Comment by timneutkens** [comment]
I'm not sure what you mean, can you rephrase the issue and provide clear examples of what you're trying to achieve?

**Comment by romainquellec** [comment]
Sorry.

For example, for this boilerplate, after build phase : index.ts and lambda.ts is transform in index.js and lambda.js, as (lambda.js) :
```
const awsServerlessExpress = require("aws-serverless-express");
const compression = require("compression");
const express = require("express");
const path = require("path");
const pathMatch = require("path-match");
const raven = require("raven");
const ravenLambdaWrapper = require("serverless-sentry-lib");
const { parse } = require("url");
// setup Express and hook up Next.js handler
const app = express();
app.use(compression());
const route = pathMatch();
const matches = [
    { route: route("/workshops/:id/edit"), page: "/workshop-edit" },
    { route: route("/workshops/new"), page: "/workshop-new" },
    { route: route("/workshops/:id"), page: "/workshop" },
];
const binaryMimeTypes = ['*/*'];
app.use("/_next/static", express.static(path.join(__dirname, "./static")));
app.get('/', require('./serverless/pages/index').render);
app.get('*', (req, res) => {
    const parsedUrl = parse(req.url, true);
    const { pathname, query } = parsedUrl;
    let hasMatch = false;
    for (const match of matches) {
        const params = match.route(pathname);
        if (params) {
            try {
                require(`./serverless/pages${pathname}`).render(req, res, match.page, Object.assign(params, query));
            }
            catch (err) {
                require('./serverless/pages/_error').render(req, res, match.page, Object.assign(params, query));
            }
            hasMatch = true;
            break;
        }
    }
    if (!hasMatch) {
        try {
            require(`./serverless/pages${pathname}`).render(req, res, parsedUrl);
        }
        catch (err) {
            require('./serverless/pages/_error').render(req, res, parsedUrl);
        }
    }
});
// 404 handler
app.get("*", require('./serverless/pages/_error').render);
const server = awsServerlessExpress.createServer(app, null, binaryMimeTypes);
const lambda = (event, context) => awsServerlessExpress.proxy(server, event, context);
const ravenHandler = ravenLambdaWrapper.handler(raven, lambda);
// export the wrapped handler for the Lambda runtime
exports.handler = ravenHandler;
//# sourceMappingURL=lambda.js.map
```

When the lambda is executed, "require("aws-serverless-express");" is problematic as dependencies are not included in the lambda (or inlined, as in pages/something.js).

I know I can do that in a seperated process with another webpack, but maybe next can take care of this, as is it done on "/pages".

**Comment by romainquellec** [comment]
as far as I know, alternatives solutions are : 
- https://github.com/liady/webpack-node-externals
- https://github.com/goldwasserexchange/serverless-plugin-webpack
- https://github.com/serverless-heaven

**Comment by romainquellec** [comment]
no idea ?

**Comment by romainquellec** [comment]
It's not resolved but I'm adding a step after build :
https://github.com/romainquellec/boilerplate/commit/f06eef6ed39ab704428b953468656eb40a6f081d

Maybe, there is a way to do that with next ?
